build_runner/lib/src/asset/finalized_reader.dart

```diff
@@ -19,6 +21,20 @@ class FinalizedReader implements AssetReader {
     this._assetGraph,
   );
 
+  /// Returns a reason why [id] is not readable, or null if it is readable.
+  Future<UnreadableAssetReason> canReadReason(AssetId id) async {
```

Comment:
This name implies the opposite of what it returns to me :D, maybe `unreadableReason`?

Reply:
+1